aten/src/ATen/native/ScatterGatherShapeChecks.h

```diff
@@ -10,7 +10,7 @@ namespace {
 
 // Used for `gather`-like methods
 // Test:
-// 1. index.size(d) == self.size(d) for all d != dim
+// 1. index.size(d) <= self.size(d) for all d != dim
```

Comment:
Maybe add a comment in the code to specify what we do in the `<` case: extra outputs will be ignored.